pkg/api/handlers/compat/exec.go

```diff
@@ -58,6 +59,7 @@ func ExecCreateHandler(w http.ResponseWriter, r *http.Request) {
 	libpodConfig.WorkDir = input.WorkingDir
 	libpodConfig.Privileged = input.Privileged
 	libpodConfig.User = input.User
+	util.ExecAddTERM(input.Tty && !ctr.Terminal(), ctr.Env(), libpodConfig.Environment)
```

Comment:
I don't see why this should ever depend on `ctr.Terminal()`? Either exec is started with tty or not, I don't see why the container option matters in this case.